R/zchunk_L202.Ccoef.R

```diff
@@ -30,43 +32,63 @@ module_energy_L202.Ccoef_DISABLED <- function(command, ...) {
     L102.Ccoef_kgCGJ_R_F_Yh <- get_data(all_data, "L102.Ccoef_kgCGJ_R_F_Yh")
 
     # ===================================================
-    # TRANSLATED PROCESSING CODE GOES HERE...
-    #
-    # If you find a mistake/thing to update in the old code and
-    # fixing it will change the output data, causing the tests to fail,
-    # (i) open an issue on GitHub, (ii) consult with colleagues, and
-    # then (iii) code a fix:
-    #
-    # if(OLD_DATA_SYSTEM_BEHAVIOR) {
-    #   ... code that replicates old, incorrect behavior
-    # } else {
-    #   ... new code with a fix
-    # }
-    #
-    #
-    # NOTE: there are 'match' calls in this code. You probably want to use left_join_error_no_match
-    # For more information, see https://github.com/JGCRI/gcamdata/wiki/Name-That-Function
+    # Carbon contents of fuels in all regions
+    A_PrimaryFuelCCoef %>%
+      write_to_all_regions(c(LEVEL2_DATA_NAMES[["CarbonCoef"]]), GCAM_region_names = GCAM_region_names, has_traded = TRUE) -> L202.CarbonCoef
+
+
+    # For regions with no agricultural and land use sector (Taiwan),
+    # need to remove the passthrough supplysectors for first-gen biofuels
+    ag_en <- c( "regional corn for ethanol", "regional sugar for ethanol", "regional biomassOil" )
+    L202.CarbonCoef %>%
+      filter(!region %in% aglu.NO_AGLU_REGIONS | !PrimaryFuelCO2Coef.name %in% ag_en) -> L202.CarbonCoef
+
+    # if emissions.USE_GCAM3_CCOEFS = 1, using exogenous C coefs from GCAM 3.0
+    if(emissions.USE_GCAM3_CCOEFS == 0) {
+      L202.CarbonCoef %>%
+        left_join(fuel_to_Ccoef, by = "PrimaryFuelCO2Coef.name") -> L202.CarbonCoef
+      if(emissions.USE_GLOBAL_CCOEFS == 1) {
+        # Using global average emissions coefficients based on CDIAC inventory and IEA energy balances
+        L102.Ccoef_kgCGJ_F_Yh %>%
+          filter(year %in% HISTORICAL_YEARS) %>%
+          filter(year == emissions.INVENTORY_MATCH_YEAR) -> L202.Ccoef_kgCGJ_F_fby
+
+        L202.CarbonCoef %>%
+          left_join(L202.Ccoef_kgCGJ_F_fby, by = "fuel") %>%
+          mutate(PrimaryFuelCO2Coef = if_else(!is.na(fuel), round(value, emissions.DIGITS_CO2COEF), PrimaryFuelCO2Coef)) %>%
+          select(one_of(LEVEL2_DATA_NAMES[["CarbonCoef"]])) -> L202.CarbonCoef
+      }
+      if(emissions.USE_GLOBAL_CCOEFS == 0) {
+        # Using region-specific average emissions coefficients based on CDIAC inventory and IEA energy balances
+        L102.Ccoef_kgCGJ_R_F_Yh %>%
+          filter(year %in% HISTORICAL_YEARS) %>%
+          filter(year == emissions.INVENTORY_MATCH_YEAR) %>%
+          left_join(GCAM_region_names, by = "GCAM_region_ID") -> L202.Ccoef_kgCGJ_R_F_fby
+
+        L202.CarbonCoef %>%
+          left_join(L202.Ccoef_kgCGJ_R_F_fby, by = c("fuel", "region")) %>%
+          mutate(PrimaryFuelCO2Coef = if_else(!is.na(fuel), round(value, emissions.DIGITS_CO2COEF), PrimaryFuelCO2Coef)) %>%
+          select(one_of(LEVEL2_DATA_NAMES[["CarbonCoef"]])) -> L202.CarbonCoef
+      }
```

Comment:
`} else stop("Unknown emissions.USE_GCAM3_CCOEFS value ", emissions.USE_GCAM3_CCOEFS)`

Reply:
@bpbond quick question on this -- if we do want to replicate the old data system, we wouldn't want this to stop if USE_GCAM3_CCOEFS = 1. Did you want me to add a stop / warning for cases that are not 0 or 1?

Reply:
That’s what I was thinking, yes. Thanks.